shared/routing/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+use std::collections::HashMap;
+
+#[derive(Debug)]
+enum PathSegment {
+    Static(String),
+    Param(String),
+}
+
+#[derive(Debug)]
+struct Path {
+    segments: Vec<PathSegment>,
+    has_trailing_splat: bool,
+}
```

Comment:
Would you actually want to use approach for the ingest-router as well? My understanding of the relay routes is that they would all be exact match, (e.g. they just look like `/projectconfig/` etc) and we don't need to break down the path into segments, support trailing splat or dynamic or anything like that. We could just do an exact string comparison and skip all this

Reply:
That is valid point. I am refactoring it with the proxy use case for now. But the intention is that i can add another variant to the routing match which can do exact matching on the whole path for ingest-router. Its on my to-do to try it out and see what changes would be needed to support it.